src/components/map/MapLibre.tsx

```diff
@@ -0,0 +1,315 @@
+        rotateEnabled={false}
+        style={[styles.map, mapStyle]}
+        zoomEnabled
+      >
+        <Camera
+          centerCoordinate={[initialRegion.longitude, initialRegion.latitude]}
+          followUserLocation={followsUserLocation}
+          followUserMode={UserTrackingMode.FollowWithHeading}
+          minZoomLevel={minZoom}
+          ref={cameraRef}
+          zoomLevel={initialZoomLevel}
+        />
+
+        <UserLocation
+          androidRenderMode="compass"
+          renderMode={UserLocationRenderMode.Native}
+          showsUserHeadingIndicator
+          visible={showsUserLocation}
+        />
+
+        <Images images={markerImages} />
+
+        <ShapeSource
+          id="benches"
+          ref={shapeSourceRef}
+          shape={featureCollection(
+            locations?.map((location) =>
+              point([location.position.longitude, location.position.latitude], { ...location })
+            )
+          )}
+          onPress={handleClusterPress}
+          cluster
+          clusterRadius={50}
+          clusterMaxZoomLevel={clusterMaxZoom}
+          clusterProperties={clusterProperties}
+        >
+          <CircleLayer
+            id="cluster-shadow"
+            filter={['has', 'point_count']}
+            style={{ ...layerStyles.clusteredCircleShadow, circlePitchAlignment: 'map' }}
+          />
+
+          <SymbolLayer
+            id="single-icon"
+            filter={['!', ['has', 'point_count']]}
+            style={{ ...layerStyles.singleIcon, iconImage: ['get', 'iconName'] }}
+          />
+
+          <CircleLayer
+            id="cluster"
+            filter={['has', 'point_count']}
+            style={{
+              ...layerStyles.clusteredCircle,
+              circlePitchAlignment: 'map',
+              circleColor: clusterCircleColor
+            }}
+          />
+
+          <SymbolLayer
+            id="cluster-count"
+            style={{
+              ...layerStyles.clusterCount,
+              textField: ['format', ['concat', ['get', 'point_count']]],
+              textPitchAlignment: 'map'
+            }}
+          />
+        </ShapeSource>
+
+        {selectedFeature && calloutTextEnabled && (
+          <MarkerView coordinate={selectedFeature.geometry.coordinates}>
+            <CustomCallout feature={selectedFeature} />
+          </MarkerView>
+        )}
+      </MapView>
+
+      {isMyLocationButtonVisible && (
+        <View style={styles.myLocationContainer}>
+          <TouchableOpacity
+            accessibilityLabel={`${texts.components.map} ${a11yLabel.button}`}
+            onPress={() => {
+              setFollowsUserLocation(true);
+              setTimeout(() => setFollowsUserLocation(false), 2000);
+            }}
+            style={styles.buttons}
+          >
+            <Icon.GPS size={normalize(18)} />
+          </TouchableOpacity>
+        </View>
+      )}
+    </View>
+  );
+};
+/* eslint-enable complexity */
+
+const styles = StyleSheet.create({
```

Comment:
I would have preferred an alphabetical sorting of styles

Reply:
done